pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/operands/TransformOperand.java

```diff
@@ -18,26 +18,123 @@
  */
 package org.apache.pinot.query.runtime.operator.operands;
 
+
+import com.google.common.base.Preconditions;
+import java.util.List;
 import org.apache.pinot.common.utils.DataSchema;
 import org.apache.pinot.query.planner.logical.RexExpression;
+import org.apache.pinot.query.runtime.operator.utils.FunctionInvokeUtils;
+import org.apache.pinot.query.runtime.operator.utils.OperatorUtils;
 
 
 public abstract class TransformOperand {
   protected String _resultName;
   protected DataSchema.ColumnDataType _resultType;
 
-  public static TransformOperand toTransformOperand(RexExpression rexExpression, DataSchema dataSchema) {
+  public static TransformOperand toTransformOperand(RexExpression rexExpression, DataSchema inputDataSchema) {
     if (rexExpression instanceof RexExpression.InputRef) {
-      return new ReferenceOperand((RexExpression.InputRef) rexExpression, dataSchema);
+      return new ReferenceOperand((RexExpression.InputRef) rexExpression, inputDataSchema);
     } else if (rexExpression instanceof RexExpression.FunctionCall) {
-      return new FunctionOperand((RexExpression.FunctionCall) rexExpression, dataSchema);
+      return toTransformOperand((RexExpression.FunctionCall) rexExpression, inputDataSchema);
     } else if (rexExpression instanceof RexExpression.Literal) {
       return new LiteralOperand((RexExpression.Literal) rexExpression);
     } else {
       throw new UnsupportedOperationException("Unsupported RexExpression: " + rexExpression);
     }
   }
 
+  @SuppressWarnings({"ConstantConditions", "rawtypes", "unchecked"})
+  private static TransformOperand toTransformOperand(RexExpression.FunctionCall functionCall,
+      DataSchema inputDataSchema) {
+    final List<RexExpression> functionOperands = functionCall.getFunctionOperands();
+    int operandSize = functionOperands.size();
+    switch (OperatorUtils.canonicalizeFunctionName(functionCall.getFunctionName())) {
+      case "AND":
+        Preconditions.checkState(operandSize >= 2, "AND takes >=2 argument, passed in argument size:" + operandSize);
+        return new FilterOperand.And(functionOperands, inputDataSchema);
+      case "OR":
+        Preconditions.checkState(operandSize >= 2, "OR takes >=2 argument, passed in argument size:" + operandSize);
+        return new FilterOperand.Or(functionOperands, inputDataSchema);
+      case "NOT":
+        Preconditions.checkState(operandSize == 1, "NOT takes one argument, passed in argument size:" + operandSize);
+        return new FilterOperand.Not(functionOperands.get(0), inputDataSchema);
+      case "equals":
+        return new FilterOperand.Predicate(functionOperands, inputDataSchema) {
+          @Override
+          public Boolean apply(Object[] row) {
```

Comment:
Not really related to this PR but a general thought - in future, can this done in columnar manner / column at a type for a batch of records ?

Reply:
yes. we are considering this but will probably not get to it until next year :-/